File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    answer=0
    for i in range(len(s)):
        newString = s[i:]+s[:i]
        stack=[]
        for j in newString:
            if len(stack) == 0 and ( (j == ']') or (j == '}') or (j == ')')):
                stack.append('1')
                break
        
            if j in ['[','{','(']:
                stack.append(j)
                continue

            if len(stack) != 0:
                if j == ']' and stack[-1] == '[':
                    stack.pop()
                elif j == '}' and stack[-1] == '{':
                    stack.pop()
                elif j == ')' and stack[-1] == '(':
                    stack.pop()
                else:
                    stack.append('1')
                    break
            else:
                stack.append('1')
                break
        if len(stack) == 0:
            answer+=1
    return answer

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
const validateBracket = (array, pair) => {
    const copyArray = [...array]
    const validate = []

    while(copyArray.length !== 0) {
        const val = copyArray.shift()
        if(!!pair[val]) {
            if (validate.at(-1) === pair[val]) validate.pop()
            else return false
        }
        else validate.push(val)
    }
    
    return validate.length ? false : true
}

function solution(s) {
    let answer = 0
    const sArray = s.split('')
    const len = sArray.length
    const pair = {')':'(', ']':'[', '}':'{'}
    
    if(validateBracket(sArray, pair)) answer++
    for(let i = 1; i < len; i++) {
        const val = sArray.shift()
        sArray.push(val)
        if(validateBracket(sArray, pair)) answer++
    }
    
    return answer
}